File: Neetcode Questions/Math & Geometry/set-matrix-zeroes.cpp

// https://leetcode.com/problems/set-matrix-zeroes/

class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        
        int rows = matrix.size();
        if(rows == 0) return;
        int cols = matrix[0].size();
        vector<int> rowSet(rows), colSet(cols);

        for(int i = 0; i < rows; i++){
            for(int j = 0; j < cols; j++){
                if(matrix[i][j] == 0){
                    rowSet[i] = 1;
                    colSet[j] = 1;
                }
            }
        }

        for(int i = 0; i < rows; i++){
            for(int j = 0; j < cols; j++){
                if(rowSet[i] == 1 || colSet[j] == 1){
                    matrix[i][j] = 0;
                }
            }
        }

        return;
    }
};

File: Neetcode-150-js/Math & Geometry/set-matrix-zeroes.js

// https://leetcode.com/problems/set-matrix-zeroes/

/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var setZeroes = function(matrix) {
    let rowSize = matrix.length;
    let colSize = matrix[0].length;

    let rows = Array.from({length : rowSize}, () => 0);
    let cols = Array.from({length : colSize}, () => 0);
    
    for(let i = 0; i < rowSize; i++){
        for(let j = 0; j < colSize; j++){
            if(matrix[i][j] === 0){
                rows[i] = 1;
                cols[j] = 1;
            }
        }
    }

    for(let i = 0; i < rowSize; i++){
        for(let j = 0; j < colSize; j++){
            if(rows[i] === 1 || cols[j] === 1){
                matrix[i][j] = 0;
            }
        }
    }
};